test/support/rtmp_source_test_pipeline.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Membrane.RTMP.Source.Test.Pipeline do
+  @moduledoc false
+  use Membrane.Pipeline
+
+  import Membrane.ParentSpec
+
+  alias Membrane.Testing
+
+  @impl true
+  def handle_init(socket: socket, test_process: test_process) do
+    Process.register(self(), __MODULE__)
+
+    spec = %Membrane.ParentSpec{
+      children: [
+        src: %Membrane.RTMP.SourceBin{socket: socket},
+        audio_sink: Testing.Sink,
+        video_sink: Testing.Sink
+      ],
+      links: [
+        link(:src) |> via_out(:audio) |> to(:audio_sink),
+        link(:src) |> via_out(:video) |> to(:video_sink)
+      ]
+    }
+
+    send(test_process, {:pipeline_started, self()})
+
+    {{:ok, [spec: spec, playback: :playing]}, %{socket: socket}}
+  end
+
+  @impl true
+  def handle_notification({:rtmp_source_initialized, source}, :src, _ctx, state) do
+    send(self(), {:rtmp_source_initialized, source})
+
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_other({:rtmp_source_initialized, source}, _ctx, state) do
+    case :gen_tcp.controlling_process(state.socket, source) do
```

Comment:
I think we need a helper function inside the `SourceBin` that will hide the the raw `:gen_tcp.controlling_process`, it should just take the source payload and look something like 
```elixir
Membrane.RTMP.SourceBin.pass_control(state.socket, source)
```
Would be probably more friendly for a user than having to do that erlang style.

Reply:
Or have some other better mechanism, not sure how it should look though.

Reply:
We will stick with the basic wrapper for `:gen_tcp:controlling_process`.